integration_tests/src/main/python/array_test.py

```diff
@@ -249,3 +249,63 @@ def test_sql_array_scalars(query):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : spark.sql('SELECT {}'.format(query)),
             conf={'spark.sql.legacy.allowNegativeScaleOfDecimal': 'true'})
+
+
```

Comment:
It is much more reasonable to move the tests to `cast_test.py`, I agree with you! 

However, you may find that tests for casting struct to string are all in `struct_test.py` instead of `cast_test.py`:
https://github.com/NVIDIA/spark-rapids/blob/877c1b275bc8aa3df49386c882355f089f08ad17/integration_tests/src/main/python/struct_test.py#L75-L166

Should we be consistent with it?

Reply:
Yes we should, and the struct tests should probably move too at some point. If you want to file a follow on issue you can. But this is a nit so if you just want both to be a follow on issue that is fine too.